actix/src/sec-7-actor-future.md

```diff
@@ -0,0 +1,104 @@
+# ActorFuture
+
+One of the pleasant and powerful patterns of Actix is the fact that Actors can
+run on only [`Arbiter`](./sec-5-arbiter) at a time. This means that Actor
+methods like `Handler`s and `Actor::started` can access the Actor's state and
+context mutably without using synchronization primitives.
+
+However, not all events are based on external events like messages. The Actor
+itself has the ability to spawn futures on the Context, and these futures
+
+If you intend on spawning the Future on the Arbiter, you can give it access to
+the Actor and the Context by using `ActorFuture`.
+
+## Example
+
+Imagine an Actor that exists to poll data from an API. Every second, it updates
+its cache. Upon request, it responds with the most recent cached data.
+
+Note: This example isn't a recommendation of when to use this pattern. It's just
+an example of the mechanics.
+
+For the sake of this example, we'll pretend we have a function that calls out to
+the API, gets the response, and parses it.
+
+```rust
+# extern crate actix;
+# extern crate futures;
+# use actix::prelude::*;
+# use futures::Future;
+# use std::time::Duration;
+#
+# #[derive(Debug)]
+# enum Weather {
+#     Rain,
+#     Sun,
+# }
+#
+# fn get_weather() -> impl Future<Item = Weather, Error = ()> {
+#     futures::future::ok(Weather::Rain)
+# }
+#
+struct WeatherActor {
+    current_weather: Weather,
+}
+
+impl WeatherActor {
+    pub fn new() -> Self {
+        Self {
+            current_weather: Weather::Sun,
+        }
+    }
+
+    fn update_weather(&mut self, ctx: &mut <Self as Actor>::Context) {
```

Comment:
I actually don't really want this to be an instance method. The problem was if I put it in the callback on line 82, we'd be so indented that I'd have to make pretty major concessions just to have the code not require the user to scroll horizontally.